netdata/netdata
# Update ebpf dashboard

##### Summary
This is the second PR that is improving our user navigation among eBPF charts, it is bringing:

- A main menu in `netdata monitoring` -> `ebpf.plugin`
- New description for Process, Hard Disk, File System (File access and Directory Cache).
- Cleanup some content that were already moved for our documentation.

The content of the PR was already discussed with product team (@cpipilas ) and our documentation team (@DShreve2 ), and we know that `cgroups` different of `systemd` does not have specific links. We also avoided exclusive links to disks. This does not mean that in the future we will address this issues, but for the next Netdata release, we are only addressing static content.

##### Test Plan

1. Compile this branch, or only copy `web/gui/dashboard_info.js` to `/usr/share/netdata/web/dashboard_info.js`
2. Change your `/etc/netdata/ebpf.d.conf` to enable integration with `cgroup` and `apps`:
```conf
[global]
    apps = yes
    cgroups = yes
```
3. Restart netdata
4. Do a force reload on your dashboard to clean up the cache.
5. Go to `netdata monitoring` -> `ebpf.plugin` and start navigating trough links ( see additional information)

##### Additional Information

This PR was tested running `mozilla-firefox-98.0.1` and `99.0.4844.82 (Official Build) (64-bit)`.

It is not necessary to test on different kernels, because our goal with this PR is to modify only dashboard.

All the external links must open in a different tab.

<details> <summary>Grep tests</summary>
I do not know the bash knowledge of each reviewer, so I created the simplest tests possible:

```sh
# Access to cgroup documentation
bash-5.1# grep "https://learn.netdata.cloud/docs/agent/collectors/ebpf.plugin#integration-with-cgroupsplugin" web/gui/dashboard_info.js | wc -l
43
bash-5.1# grep "https://learn.netdata.cloud/docs/agent/collectors/ebpf.plugin#integration-with-cgroupsplugin" web/gui/dashboard_info.js | grep "_blank" | wc -l
43

# Access to application documentation
bash-5.1# grep "https://learn.netdata.cloud/guides/troubleshoot/monitor-debug-applications-ebpf" web/gui/dashboard_info.js | grep applications | wc -l
47
bash-5.1# grep "https://learn.netdata.cloud/guides/troubleshoot/monitor-debug-applications-ebpf" web/gui/dashboard_info.js | grep applications | grep "_blank" | wc -l
47

# Divs named ebpf_apps
bash-5.1# grep "#ebpf_apps" web/gui/dashboard_info.js | wc -l
43

# Divs named ebpf_services
bash-5.1# grep "#ebpf_services" web/gui/dashboard_info.js | wc -l
43
bash-5.1# grep "cgroup (systemd Services)" web/gui/dashboard_info.js | grep "#ebpf_services" | wc -l
43

# Access to specific links outside our dashboard
bash-5.1# grep "https://learn.netdata.cloud/docs/agent/collectors/ebpf.plugin" web/gui/dashboard_info.js | wc -l
127
bash-5.1# grep "https://learn.netdata.cloud/docs/agent/collectors/ebpf.plugin" web/gui/dashboard_info.js | grep "_blank" |  wc -l
127

# Correct number of constant usage
# We expect declaration and more two usages for cgroup
bash-5.1# for i in `grep const web/gui/dashboard_info.js | grep ebpf | cut -d= -f1 | cut -d" " -f2`; do grep -w $i web/gui/dashboard_info.js| wc -l; done
97 #ebpfChartProvides
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3
3

```
</details>

<details> <summary>For users</summary>
This PR is updating  dashboard navigation. We want avoid users clicking everywhere to find specific contents. After this PR to be merged users can find eBPF charts using the following road:

1. Access `Netdata monitoring` -> `eBPF.plugin`
2. Clink on link and go to a global chart.
3. From global chart they have links to application and systemd Services.
4. When they access systemd, they can return for global chart or go to apps.
5. When they access application, they can return for global chart or go to systemd.
</details>
